improver/wxcode/modal_code.py

```diff
@@ -151,6 +152,28 @@ def mode_aggregator(data: ndarray, axis: int) -> ndarray:
         )
         return CODE_MAX - np.squeeze(mode_result)
 
+    @staticmethod
+    def _set_blended_times(cube: Cube) -> None:
+        """Updates time coordinates so that time point is at the end of the time bounds,
+        blend_time and forecast_reference_time (if present) are set to the end of the
+        bound period and bounds are removed, and forecast_period is updated to match."""
+        cube.coord("time").points = cube.coord("time").bounds[0][-1]
+
+        for coord_name in ["blend_time", "forecast_reference_time"]:
+            if coord_name in [c.name() for c in cube.coords()]:
+                coord = cube.coord(coord_name)
+                if coord.has_bounds():
+                    coord = coord.copy(coord.bounds[0][-1])
+                    cube.replace_coord(coord)
```

Comment:
Thanks @bayliffe. As discussed, I think this is a good idea, but has the consequence of changing the weighted-blending CLI such that cycle time is not allowed when blending cycles, which would require a suite change. It would be better to do this after the 2022.1 release.